File: order-server/src/main/java/com/example/order/dto/OrderDto.java
package com.example.order.dto;

import com.example.order.entity.Order;
import lombok.*;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OrderDto {
    private Long id;
    private Long userId;
    private Order.OrderStatus status;
    private List<OrderItemDto> orderItems;

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class OrderItemDto {
        private Long productId;
        private String productName;
        private BigDecimal unitPrice;
        private Integer quantity;
    }
}

File: order-server/src/main/kotlin/com/example/order/dto/OrderDto.kt
package com.example.order.dto

import com.example.order.entity.Order
import java.math.BigDecimal

data class OrderDto(
    val id: Long? = null,
    val userId: Long,
    val status: Order.OrderStatus? = null,
    val orderItems: List<OrderItemDto>? = null
) {
    data class OrderItemDto(
        val productId: Long,
        val productName: String,
        val unitPrice: BigDecimal,
        val quantity: Int
    )
}
